test/jdk/sun/net/www/http/HttpClient/KeepAliveTest.java

```diff
@@ -0,0 +1,590 @@
+            System.out.println("HttpClient in Cache:" + httpClientCached);
+            if(httpClientCached != null) {
+                System.out.println("KeepingAlive:" + httpClientCached.isKeepingAlive());
+                System.out.println("UsingProxy:" + httpClientCached.getUsingProxy());
+                System.out.println("ProxiedHost:" + httpClientCached.getProxyHostUsed());
+                System.out.println("ProxiedPort:" + httpClientCached.getProxyPortUsed());
+                System.out.println("ProxyPortUsingSystemProperty:" + System.getProperty("http.proxyPort"));
+                System.out.println("ProxyHostUsingSystemProperty:" + System.getProperty("http.proxyHost"));
+                System.out.println("http.keepAlive.time.server=" + System.getProperty("http.keepAlive.time.server"));
+                System.out.println("http.keepAlive.time.proxy=" + System.getProperty("http.keepAlive.time.proxy"));
+
+                Class clientVectorClass = Class.forName("sun.net.www.http.ClientVector");
+                //      System.out.println("clientVectorClass=" + clientVectorClass);
+                Field napField = clientVectorClass.getDeclaredField("nap");
+                napField.setAccessible(true);
+                int napValue = (int) napField.get(clientVectorObjectInMap);
+                int actualValue = napValue / 1000;
+                //      System.out.println("nap=" + napValue / 1000);
+                System.out.printf("ExpectedOutput:%d ActualOutput:%d ", expectedValues[scenarioNumber], actualValue);
+                System.out.println();
+                if (expectedValues[scenarioNumber] != actualValue) {
+                    throw new RuntimeException(
+                            "ExpectedOutput:" + expectedValues[scenarioNumber] + " ActualOutput: " + actualValue);
+                }
+            } else {
+                //executed when value is not cached.
+                String expected = expectedValues[scenarioNumber] == 0 ? NOT_CACHED
+                    : String.valueOf(expectedValues[scenarioNumber]);
+                System.out.println("ExpectedOutput:" + expected + " ActualOutput:" + NOT_CACHED);
+                if (!expected.equalsIgnoreCase(NOT_CACHED)) {
+                    throw new RuntimeException("ExpectedOutput:" + expected + " ActualOutput:" + NOT_CACHED);
+                }
+            }
+        }
+    public static void main(String[] args) throws Exception {
+        if (args.length != 1) {
+            System.out.println("Usage:java KeepAliveTest.java <scenarioNumber>");
+            System.out.println("e.g.:java KeepAliveTest.java 1");
+            System.exit(1);
+        }
+        logger.setLevel(Level.FINEST);
+        ConsoleHandler h = new ConsoleHandler();
+        h.setLevel(Level.FINEST);
+        logger.addHandler(h);
+        KeepAliveTest keepAliveTest = new KeepAliveTest();
+        if (args.length != 0) {
+            keepAliveTest.myinit(Integer.valueOf(args[0]));
```

Comment:
It would be good to have method names that reflect what the method does. Here for instance - what `myinit` does is actually running (or starting) the given scenario. So maybe it should be called `startScenario`.

Reply:
modified name to startScenario thanks @dfuch